### PR TITLE
Update README.md: add doc

### PR DESCRIPTION
**Pull request by yhh2021**

### CODE REVIEW

**Comment by yhh2021** [comment]
> Why is this comment necessary?

TL;DR: such technique are not standard and not widely supported. Many people's time will be wasted.

Long explanation: When I saw this section, I was surprised that there are GC implementations, but then I became puzzled: what? how does it work? just create some pointers and then they GC like a charm?

I then went to these projects' page, and similarly there are no explanation. (in fact one project explains, but it is too short that I didn't see it at the first time.) I then dig into the source, and finally, oops, just a wrapper to __attribute__ stuff, which means I can not use them for the sake of cross-platform. My time is wasted, thus I add a comment to warn future visitors.

**Comment by oz123** [comment]
Ok. Since you already dug into the source code, do you know which gc implementation uses __attribute__ or is it all of them?
